File: examples/bimapi/CheckbotBimLink/CppFeaExample/BimApiFeaLink/Importers/NodeImporter.cs
using FeaApi;
using IdeaStatiCa.BimApi;
using IdeaStatiCa.BimApiLink.BimApi;
using IdeaStatiCa.BimApiLink.Importers;

namespace BimApiFeaLink.Importers
{
	public class NodeImporter : IntIdentifierImporter<IIdeaNode>
	{
		private readonly IFeaGeometryApi geometry;

		public NodeImporter(IFeaGeometryApi geometry)
		{
			this.geometry = geometry;
		}

		public override IIdeaNode Create(int id)
		{
			var v = GetLocation(id);
			return new IdeaNode(id)
			{
				Vector = v,
			};
		}

		private IdeaVector3D GetLocation(int id)
		{
			IFeaNode feaNode = geometry.GetNode(id);
			return new IdeaVector3D(feaNode.X, feaNode.Y, feaNode.Z);
		}
	}
}

File: examples/bimapi/CheckbotBimLink/CppFeaExample/CppFeaApiWrapper/NodeImporter.cpp
#include "pch.h"
#include "NodeImporter.h"

namespace CppFeaApiWrapper
{
	namespace Importers
	{
		NodeImporter::NodeImporter(ImporterContext^ context)
		{
			this->context = context;
		}

		IdeaVector3D^ NodeImporter::GetLocation(int id)
		{
			NativeFeaNode* feaNode = this->context->GetGeometry()->GetNode(id);
			IdeaStatiCa::BimApi::IdeaVector3D^ vect = gcnew IdeaVector3D(feaNode->X, feaNode->Y, feaNode->Z);
			return vect;
		}

		IIdeaNode^ NodeImporter::Create(int id)
		{
			IdeaVector3D^ v = GetLocation(id);
			IdeaNode^ node = gcnew IdeaNode(id);
			node->Vector = v;
			return node;
		}
	}
}